assets/src/edit-story/components/library/panes/media/mediaEditDialog.js

```diff
@@ -0,0 +1,210 @@
+  font-size: ${({ theme }) => theme.fonts.body1.size};
+  color: ${({ theme }) => theme.colors.bg.v11};
+`;
+
+const Input = styled.input`
+  background: ${({ theme }) => theme.colors.bg.v13};
+  border: 1px solid ${({ theme }) => theme.colors.fg.v3};
+  box-sizing: border-box;
+  border-radius: 4px;
+  font-family: ${({ theme }) => theme.fonts.input.family};
+  line-height: ${({ theme }) => theme.fonts.input.lineHeight};
+  font-size: ${({ theme }) => theme.fonts.input.size};
+  padding: 7px 10px;
+  margin-top: 20px;
+  margin-bottom: 4px;
+`;
+
+const DialogDescription = styled.div`
+  font-family: ${({ theme }) => theme.fonts.description.family};
+  line-height: ${({ theme }) => theme.fonts.description.lineHeight};
+  font-weight: ${({ theme }) => theme.fonts.description.weight};
+  font-size: ${({ theme }) => theme.fonts.description.size};
+  color: ${({ theme }) => theme.grayout};
+`;
+
+const imageDialogTitle = __('Edit Image', 'web-stories');
+const videoDialogTitle = __('Edit Video', 'web-stories');
+const imageDialogDescription = __(
+  'Describe the appearance and function of an image. Leave empty if the image is purely decorative',
+  'web-stories'
+);
+const videoDialogDescription = __(
+  'Describe the appearance and function of a video. Leave empty if the video is purely decorative',
+  'web-stories'
+);
+
+/**
+ * Displays a dialog where the user can edit the selected media element.
+ *
+ * @param {Object} props Component props.
+ * @param {Object} props.resource Selected media element's resource object.
+ * @param {Object} props.onClose Callback to toggle dialog display.
+ * @return {null|*} The dialog element.
+ */
+function MediaEditDialog({ resource, onClose }) {
+  const {
+    id,
+    src,
+    title,
+    width,
+    height,
+    type,
+    alt,
+    poster,
+    mimeType,
+  } = resource;
+  const {
+    actions: { updateMedia },
+  } = useAPI();
+  const {
+    actions: { updateMediaElement },
+  } = UseMedia();
+  const { showSnackbar } = useSnackbar();
+  const [altText, setAltText] = useState(alt);
+
+  const handleAltTextChange = useCallback((evt) => {
+    setAltText(evt.target.value);
+  }, []);
+
+  const updateMediaItem = useCallback(async () => {
+    try {
+      // Update server.
+      await updateMedia(id, { alt_text: altText });
+      // Update internal state.
+      updateMediaElement({ id, alt: altText });
+      onClose();
+    } catch (err) {
+      showSnackbar({
+        message: __('Failed to update, please try again.', 'web-stories'),
+      });
+    }
+  }, [altText, id, onClose, showSnackbar, updateMedia, updateMediaElement]);
+
+  return (
+    <Dialog
+      open={true}
+      onClose={onClose}
+      title={type == 'image' ? imageDialogTitle : videoDialogTitle}
+      actions={
+        <>
+          <Plain onClick={onClose}>{__('Cancel', 'web-stories')}</Plain>
+          <Plain onClick={updateMediaItem}>{__('Save', 'web-stories')}</Plain>
```

Comment:
This one should be a primary button, not plain

Reply:
https://github.com/google/web-stories-wp/pull/1656#discussion_r431052042

Reply:
Created #2104 for this.